app/src/tooloptionwidget.cpp

```diff
@@ -151,6 +147,18 @@ void ToolOptionWidget::onToolPropertyChanged(ToolType, ToolPropertyType ePropert
 
 void ToolOptionWidget::setVisibility(BaseTool* tool)
 {
+    if (tool->type() == BUCKET) {
+        disableAllOptions();
+        if (mBucketOptionsWidget == nullptr) {
+            mBucketOptionsWidget = new BucketOptionsWidget(editor(), this);
```

Comment:
Personally, I prefer to create the widget at the initial stage. Especially when the widget will not be destroyed until the application ends.